### PR TITLE
Awaiting download, should make sleep obsolete

### PR DESCRIPTION
**Pull request by JKamsker**
Please check if we now still needthe static sleep inbetween

In my opinion thats not the case

### CODE REVIEW

**Comment by richardbang83** [comment]
How is working on you? 
Based on last commit, if i dont put the sleep btw them, the subs's name is broken due to, i guess,confliction with next downloading file.
It used the method, chrome download.
I need to validate how is your commit working whitch downloads a url directly.

**Comment by JKamsker** [comment]
It works for me. It basically resumes execution when the download has finished. So there should not be 2 simultaneous downloads anymore

**Comment by JKamsker** [comment]
@richardbang83 Can you check it out if it works for you? (& remove the extra delays inbetween, leave the last sleep lol)

**Review by richardbang83 — COMMENTED** [review]
Please check my commit.

File: script.js [hunk 1]
```diff
@@ -466,9 +463,9 @@ const downloadCourse = async (courseJSON, startingVideoId) => {
 				chrome.storage.sync.set({ Status: "Downloading..." }, undefined);
 				await downloadSubs(subsURL, filePath_subs);
 				// wait for downloading completed
-				await sleep(1000);
+				//await sleep(1000);
 
-				await downloadVideo(videoURL, filePath);
+			    downloadVideo(videoURL, filePath);
```
Comment (JKamsker) [inline script.js:468]: Why did you remove the await? We need to know wether the download worked or not to proper handle exceptions

Comment (richardbang83) [reply script.js:468]: That is for not waiting for completion when it takes longer than expected due to any reason.
The reason why I put the default delay after download is to avoid file name's crash. 
If we download the subs synchronously, don't have to worry about the video. 
It already has sleep at last.
Anyway, if we use await there, what exception do you want to handle ?

Comment (JKamsker) [reply script.js:468]: Ah ok
Maybe the server rejects our request...then we'd have to retry.
Do you know why the names sometimes fuck up?
